TESTS/mbed_drivers/timerevent/main.cpp

```diff
@@ -0,0 +1,352 @@
+    TEST_ASSERT(ticker_queue.present_time > prev_ts);
+    TEST_ASSERT_NULL(ticker_queue.head);
+}
+
+/** Template for tests: insert, insert_absolute
+ *
+ * Test insert
+ * Given an instance of @a TimerEvent subclass
+ * When a tiestamp is set with @a insert()
+ *     and given time elapses
+ * Then an event handler is called
+ *
+ * Test insert_absolute
+ * Given an instance of @a TimerEvent subclass
+ * When a tiestamp is set with @a insert_absolute()
+ *     and given time elapses
+ * Then an event handler is called
+ */
+template<typename T>
+void test_insert(void) {
+    T tte;
+    tte.sem_wait(0);
+
+    tte.set_future_timestamp(TEST_DELAY_US);
+    int32_t sem_slots = tte.sem_wait(TEST_DELAY_US / 1000 + 1);
+    TEST_ASSERT_EQUAL(1, sem_slots);
+
+    tte.remove();
+}
+
+/** Template for tests: remove
+ *
+ * Test remove after insert
+ * Given an instance of @a TimerEvent subclass
+ * When a tiestamp is set with @a insert()
+ *     and timestamp is removed before being reached
+ * Then the event handler is never called
+ *
+ * Test remove after insert_absolute
+ * Given an instance of @a TimerEvent subclass
+ * When a tiestamp is set with @a insert_absolute()
+ *     and timestamp is removed before being reached
+ * Then the event handler is never called
+ */
+template<typename T>
+void test_remove(void) {
+    T tte;
+    tte.sem_wait(0);
+
+    tte.set_future_timestamp(TEST_DELAY_US * 2);
+    int32_t sem_slots = tte.sem_wait(TEST_DELAY_US / 1000);
+    TEST_ASSERT_EQUAL(0, sem_slots);
+    tte.remove();
+
+    sem_slots = tte.sem_wait(TEST_DELAY_US * 2 / 1000 + 1);
+    TEST_ASSERT_EQUAL(0, sem_slots);
+}
+
+/** Test insert zero
+ *
+ * Given an instance of @a TimerEvent subclass initialized with
+ *     an empty @a ticker_queue structure
+ * When @a ticker_queue.present_time is greater than 0 us
+ *     and a tiestamp of 0 us is set with @a insert()
+ * Then object's internal event is scheduled with timestamp UINT32_MAX + 1
+ */
+void test_insert_zero(void) {
+    ticker_event_queue_t ticker_queue = { 0 };
+    ticker_interface_t ticker_interface = mock_ticker_interface;
+    ticker_interface.read = mock_ticker_read_const;
+    ticker_data_t ticker_data = {
+        .interface = &ticker_interface,
+        .queue = &ticker_queue
+    };
+    TestTimerEvent tte(&ticker_data);
+
+    tte.insert(0UL);
+    TEST_ASSERT_EQUAL_PTR(tte.get_event_addr(), ticker_queue.head);
+    TEST_ASSERT_EQUAL(1ULL << 32, ticker_queue.head->timestamp);
+    TEST_ASSERT_NULL(ticker_queue.head->next);
+
+    tte.remove();
+}
+
+/** Test insert_absolute zero
+ *
+ * Given an instance of @a TimerEvent subclass
+ * When a timestamp of 0 us is set with @a insert_absolute()
+ * Then an event handler is called instantly
+ */
+void test_insert_absolute_zero(void) {
+    TestTimerEvent tte;
+    tte.sem_wait(0);
+
+    tte.insert_absolute(0ULL);
+    int32_t sem_slots = tte.sem_wait(0);
+    TEST_ASSERT_EQUAL(1, sem_slots);
```

Comment:
This assertion will fail on _NUCLEO_F070RB_ and _EFM32GG_STK3700_ boards until #5051 is fixed.